sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -180,6 +180,15 @@ class SparkSession private(
    */
   @transient lazy val conf: RuntimeConfig = new RuntimeConfig(sessionState.conf)
 
+  /**
+   * Same as `spark.default.parallelism`, can be isolated across sessions.
+   *
+   * @since 3.1.0
+   */
+  def defaultParallelism: Int = {
```

Comment:
I'd like to not have this API, as `SparkSession` should provide high-level logical APIs, not physical ones.